### PR TITLE
[Sival]  Add the Silicon exec_env to test

### PR DESCRIPTION
**Pull request by engdoreis**
Add the Silicon exec_env to test so they can run on the nightlies.

### CODE REVIEW

**Review by jwnrt — APPROVED** [review]

File: sw/device/tests/BUILD [hunk 1]
```diff
@@ -4071,6 +4077,7 @@ opentitan_test(
     name = "rv_core_ibex_nmi_irq_test",
     srcs = ["rv_core_ibex_nmi_irq_test.c"],
     exec_env = dicts.add(
+        EARLGREY_SILICON_OWNER_ROM_EXT_ENVS,
```
Comment (nbdd0121) [inline sw/device/tests/BUILD:4080]: Why is this added?

File: sw/device/tests/BUILD [hunk 2]
```diff
@@ -5318,13 +5325,16 @@ opentitan_test(
...
     ),
-    exec_env = {
-        # This test is not compatible with the test_rom because the test_rom
-        # does not respect the exec_disabled OTP config word.
-        "//hw/top_earlgrey:fpga_cw310_sival": None,
-        "//hw/top_earlgrey:fpga_cw310_sival_rom_ext": None,
-        "//hw/top_earlgrey:fpga_cw310_rom_with_fake_keys": None,
-    },
+    exec_env = dicts.add(
+        EARLGREY_SILICON_OWNER_ROM_EXT_ENVS,
```
Comment (nbdd0121) [inline sw/device/tests/BUILD:5329]: Please avoid adding these to rv_core_ibex tests, they don't work on silicon and I'd like to avoid future merge conflicts.

**Review by nbdd0121 — CHANGES_REQUESTED** [review]

File: sw/device/tests/BUILD (hunk 2 above)
Comment (engdoreis) [reply sw/device/tests/BUILD]: Removed

**Review by nbdd0121 — APPROVED** [review]

File: sw/device/tests/pmod/BUILD [hunk 3]
```diff
@@ -21,10 +21,13 @@ opentitan_test(
             "pmod",
         ],  # Requires the PMOD::BoB.
     ),
-    exec_env = {
-        "//hw/top_earlgrey:fpga_cw310_rom_with_fake_keys": None,
-        "//hw/top_earlgrey:fpga_cw310_test_rom": None,
-    },
+    exec_env = dicts.add(
```
[the remark this answers is not in the document]
Comment (engdoreis) [reply sw/device/tests/pmod/BUILD:29]: Thanks for pointing out.
